### PR TITLE
Fix "original work" for tag filer

### PR DESCRIPTION
**Pull request by da3dsoul**
Continuation of https://github.com/ShokoAnime/ShokoServer/pull/893

### CODE REVIEW

File: Shoko.Server/Utilities/TagFilter.cs [hunk 1]
```diff
@@ -1,597 +1,607 @@
...
+                        toRemove.Add(sourceTag);
+                    }
+
+                    lock (toAdd)
+                    {
+                        toAdd.Add(newTag);
+                    }
+                }
+
+                if (_tagReplacementDictionary.TryGetValue(tag, out newTag))
```
Comment (revam) [inline Shoko.Server/Utilities/TagFilter.cs:597]: Why is the dictionary named "\_tag**Replacement**Dictionary" if it is a dictionary of tags to remove if some other tags are present?

**Comment by revam** [comment]
Besides the remark above then it LGTM.

File: Shoko.Server/Utilities/TagFilter.cs (hunk 1 above)
Comment (da3dsoul) [reply Shoko.Server/Utilities/TagFilter.cs:597]: Because I couldn't think of a good name. Meh

**Comment by revam** [comment]
Added a few more specific tests, otherwise it LGTM.

**Pull request merged**